inference-engine/src/transformations/src/transformations/common_optimizations/random_uniform_fusion.cpp

```diff
@@ -0,0 +1,161 @@
+
+#include <memory>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformFusion, "RandomUniformFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformMaxValFusion, "RandomUniformMaxValFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformMinValFusion, "RandomUniformMinValFusion", 0);
+
+ngraph::pass::RandomUniformMaxValFusion::RandomUniformMaxValFusion() {
+    MATCHER_SCOPE(RandomUniformMaxValFusion);
+    const auto data_pattern = ngraph::pattern::any_input();
+    const auto ru_min_input_pattern = ngraph::pattern::any_input();
+    const auto ru_max_input_pattern = ngraph::pattern::any_input();
+    const auto random_uniform_pattern =
+        ngraph::pattern::wrap_type<opset8::RandomUniform>({data_pattern, ru_min_input_pattern, ru_max_input_pattern},
+                                                          pattern::consumers_count(1));
+    const auto mul_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+
+    const auto convert_pattern = ngraph::pattern::wrap_type<opset8::Convert>({random_uniform_pattern});
+    const auto random_uniform_or_convert_pattern =
+        std::make_shared<pattern::op::Or>(OutputVector{random_uniform_pattern, convert_pattern});
+
+    const auto mul_pattern =
+        ngraph::pattern::wrap_type<opset8::Multiply>({random_uniform_or_convert_pattern, mul_const_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto data = pattern_map.at(data_pattern);
+        const auto random_uniform = pattern_map.at(random_uniform_pattern);
+        const auto mul_constant = pattern_map.at(mul_const_pattern);
+        const auto ru = std::dynamic_pointer_cast<opset8::RandomUniform>(random_uniform.get_node_shared_ptr());
+        if (!ru)
+            return false;
+
+        const auto mul_const = std::dynamic_pointer_cast<opset8::Constant>(mul_constant.get_node_shared_ptr());
+        if (!mul_const)
+            return false;
+
+        auto const_shape = mul_const->get_shape();
+        size_t const_shape_size = shape_size(const_shape);
+
+        if (const_shape_size != 1)
+            return false;
+        const auto value = mul_const->cast_vector<double>();
+        auto new_mul_const = op::Constant::create(ru->get_out_type(), Shape{}, value);
+
+        if (pattern_map.count(convert_pattern)) {
+            const auto& mul = pattern_map.at(mul_pattern);
+            const auto ml = std::dynamic_pointer_cast<opset8::Multiply>(mul.get_node_shared_ptr());
+            if (!ml)
+                return false;
+
+            const auto& convert = pattern_map.at(convert_pattern);
+            const auto cvt = std::dynamic_pointer_cast<opset8::Convert>(convert.get_node_shared_ptr());
```

Comment:
please check that `Convert` destination type is floating-point

Reply:
Done.